src/login/logind-dbus.c

```diff
@@ -1994,6 +1994,28 @@ static int strdup_job(sd_bus_message *reply, char **ret) {
         return 0;
 }
 
+static void cleanup_wakeup_count(void) {
+        if (access("/run/systemd/system/wakeup_count", F_OK))
+                unlink("/run/systemd/system/wakeup_count");
+}
+
+static int save_wakeup_count(void) {
+        _cleanup_free_ char *buf = NULL;
+        int r;
+
+        r = read_full_virtual_file("/run/systemd/system/wakeup_count", &buf, NULL);
+        if (r < 0) {
+                cleanup_wakeup_count();
+                return log_debug_errno(r, "Unable to read wakeup count: %m");
+        }
+
+        r = write_string_file("/sys/power/wakeup_count", buf, WRITE_STRING_FILE_ATOMIC);
```

Comment:
OK, what does this do? This needs to be explained in the commit message at least. A short comment here would be nice too.

Oh, and the commit message says "Save the wakeup count to /run/systemd/system/wakeup_count", is the code backwards?